apps/shell/tash_command.c

```diff
@@ -638,18 +641,25 @@ bool tash_do_autocomplete(char *cmd, int *pos, bool double_tab)
 int tash_cmd_install(const char *str, TASH_CMD_CALLBACK cb, int thread_exec)
 {
 	int cmd_idx;
-	static int fail_cmd_count = 0;
-
-	if (TASH_MAX_COMMANDS == tash_cmds_info.count) {
-		printf("Allowed Max tash cmds: %d and Current tash cmd count: %d\n",
-				TASH_MAX_COMMANDS, tash_cmds_info.count + ++fail_cmd_count);
-		printf("Couldn't install cmd: (%s), Refer CONFIG_TASH_MAX_COMMANDS\n", str);
-		return -1;				/* MAX cmd count reached */
-	}
+	struct tash_cmd_s *new_cmd_buff;
 
 	/* Lock mutex */
 	pthread_mutex_lock(&tash_cmds_info.tmutex);
 
+	/* allocate the cmd buffer */
+	if (tash_cmds_info.count == 0) {
```

Comment:
If command is already registered, we don't need to allocate the new.
So I think we'd better check the existence(line 664 to 669) before allocation or free the memory if already registered.